Masking policy: failed to parse result of describe: sql: expected 12 destination arguments in Scan, not 11

### Terraform CLI and Provider Versions

I tested on 0.68.1 and above

### Terraform Configuration

```terraform
resource "snowflake_table_column_masking_policy_application" "application" {

  table          = "table_test"
  column         = "secret"
  masking_policy = snowflake_masking_policy.policy.qualified_name
}
```


### Expected Behavior

Plan/Apply the masking policy on the column "secret" on the "table_test"

### Actual Behavior

Error: failed to parse result of describe: sql: expected 12 destination arguments in Scan, not 11

### Steps to Reproduce

1. `terraform plan`


### How much impact is this issue causing?

High

### Logs

_No response_

### Additional Information

The snowflake_masking_policy works fine. it's the snowflake_table_column_masking_policy_application that seems to be in error

Hey @sfc-gh-asawicki , 
Here is the specific statement that gives out an error

`2024-01-17T11:51:42.305+0100 [INFO]  provider.terraform-provider-snowflake_v0.68.1: 2024/01/17 11:51:42 [DEBUG] sql-conn-query [query DESCRIBE TABLE "RAW"."XXXXXXXX"."USERS" TYPE = COLUMNS; err <nil> duration 54.23575ms args {}]: timestamp=2024-01-17T11:51:42.305+0100
2024-01-17T11:51:42.306+0100 [INFO]  provider.terraform-provider-snowflake_v0.68.1: 2024/01/17 11:51:42 [DEBUG] sql-rows-next [err <nil> duration 12.458µs]: timestamp=2024-01-17T11:51:42.305+0100
2024-01-17T11:51:42.306+0100 [ERROR] provider.terraform-provider-snowflake_v0.68.1: Response contains error diagnostic: @caller=github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/internal/diag/diagnostics.go:55 @module=sdk.proto diagnostic_summary="failed to parse result of describe: sql: expected 12 destination arguments in Scan, not 11" tf_req_id=b8f0cb73-f13e-9af2-3c87-a65c658d50e1 tf_resource_type=snowflake_table_column_masking_policy_application tf_rpc=ReadResource diagnostic_detail= diagnostic_severity=ERROR tf_proto_version=5.3 tf_provider_addr=registry.terraform.io/Snowflake-Labs/snowflake timestamp=2024-01-17T11:51:42.306+0100
`
No errors before. I wanna avoid posting the full debug log because it contains credentials. I can anonymize them however if this is not enough.

Thank you very much for your help

--- a/pkg/snowflake/masking_policy_application.go
+++ b/pkg/snowflake/masking_policy_application.go
@@ -38,10 +38,10 @@ func (m *TableColumnMaskingPolicyApplicationManager) Read(x *TableColumnMaskingP
 }
 
 func (m *TableColumnMaskingPolicyApplicationManager) Parse(rows *sql.Rows, column string) (string, error) {
-	var name, sqlType, kind, null, defaultValue, primaryKey, uniqueKey, check, expression, comment, policyName sql.NullString
+	var name, sqlType, kind, null, defaultValue, primaryKey, uniqueKey, check, expression, comment, policyName, privacyDomain sql.NullString
 
 	for rows.Next() {
-		if err := rows.Scan(&name, &sqlType, &kind, &null, &defaultValue, &primaryKey, &uniqueKey, &check, &expression, &comment, &policyName); err != nil {
+		if err := rows.Scan(&name, &sqlType, &kind, &null, &defaultValue, &primaryKey, &uniqueKey, &check, &expression, &comment, &policyName, &privacyDomain); err != nil {
 			return "", err
 		}
 

